### PR TITLE
Updating to use cli

### LINKED ISSUE #3908: Change to build to use dotnet
At a minimum, you have to upgrade a few scripts and project.json files just like here: aspnet/Configuration@9cfd22f

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes #3908

### CODE REVIEW

File: NuGet.config [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <configuration>
   <packageSources>
-    <add key="AspNetVNext" value="https://www.myget.org/F/aspnetcidev/api/v3/index.json" />
+    <add key="AspNetVNext" value="https://www.myget.org/F/aspnetvolatiledev/api/v3/index.json" />
```
Comment (pranavkm) [inline NuGet.config]: Will undo prior to merging this in.

File: src/Microsoft.AspNetCore.Mvc.Core/Infrastructure/DependencyContextAssemblyProvider.cs [hunk 2]
```diff
@@ -6,12 +6,34 @@
...
+        /// </summary>
+        /// <param name="environment">The <see cref="IApplicationEnvironment"/>.</param>
+        public DependencyContextAssemblyProvider(IApplicationEnvironment environment)
+        {
+#if NET451
+            var applicationAssembly = Assembly.Load(environment.ApplicationName);
+#else
+            var applicationAssembly = Assembly.Load(new AssemblyName(environment.ApplicationName));
+#endif
+            _dependencyContext = DependencyContext.Load(applicationAssembly);
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Core/Infrastructure/DependencyContextAssemblyProvider.cs]: cc @pakrym \ @davidfowl

File: test/Microsoft.AspNetCore.Mvc.DataAnnotations.Test/project.json [hunk 3]
```diff
@@ -16,6 +16,7 @@
     "Microsoft.NETCore.Platforms": "1.0.1-*",
     "xunit.runner.aspnet": "2.0.0-aspnet-*"
```
Comment (victorhurdugaci) [inline test/Microsoft.AspNetCore.Mvc.DataAnnotations.Test/project.json]: Do you still need this? (applies to the other projects too)

File: test/WebSites/ApiExplorerWebSite/project.json [hunk 4]
```diff
@@ -3,18 +3,23 @@
...
-    "Microsoft.AspNetCore.StaticFiles": "1.0.0-*",
-    "Microsoft.NETCore.Platforms": "1.0.1-*"
+    "Microsoft.AspNetCore.StaticFiles": "1.0.0-*"
   },
   "frameworks": {
-    "dnx451": {},
-    "dnxcore50": {}
+    "net451": {},
+    "dnxcore50": {
+      "imports": "portable-net451+win8",
```
Comment (victorhurdugaci) [inline test/WebSites/ApiExplorerWebSite/project.json]: Why do you need this?

File: test/Microsoft.AspNetCore.Mvc.DataAnnotations.Test/project.json (hunk 3 above)
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.DataAnnotations.Test/project.json]: Wouldn't you need this for coreclr? I haven't switched the dnxcore tests to use `dotnet test` as yet.

Comment (victorhurdugaci) [inline test/Microsoft.AspNetCore.Mvc.DataAnnotations.Test/project.json]: Yes we did (for most of the repos). @kichalla has more info

Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.DataAnnotations.Test/project.json]: Sure, which we'll do it in a follow up. This one gets net451 to build and run tests using dotnet.

File: src/Microsoft.AspNetCore.Mvc.Core/Infrastructure/DependencyContextAssemblyProvider.cs (hunk 2 above)
Comment (dougbu) [inline src/Microsoft.AspNetCore.Mvc.Core/Infrastructure/DependencyContextAssemblyProvider.cs]: @pranavkm this code is repeated a few times. Can it be placed somewhere and reused? May need to change this a few more times before we're done.

Comment (davidfowl) [inline src/Microsoft.AspNetCore.Mvc.Core/Infrastructure/DependencyContextAssemblyProvider.cs]: Yas

File: test/WebSites/ApiExplorerWebSite/project.json (hunk 4 above)
Comment (pranavkm) [inline test/WebSites/ApiExplorerWebSite/project.json]: We consume a portable class library (one of the CodeAnalysis packages) and consequently have to do this.

File: src/Microsoft.AspNetCore.Mvc.Core/Infrastructure/DependencyContextAssemblyProvider.cs (hunk 2 above)
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Core/Infrastructure/DependencyContextAssemblyProvider.cs]: Could we move this to DependencyModel?

Comment (pakrym) [inline src/Microsoft.AspNetCore.Mvc.Core/Infrastructure/DependencyContextAssemblyProvider.cs]: I guess we can change default logic to use `AssemblyName(environment.ApplicationName)` by default in dependency context. It default to same anyway.

**Pull request closed without merging**